streams/src/main/java/org/apache/kafka/streams/kstream/internals/StatefulRepartitionNode.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+import org.apache.kafka.streams.processor.internals.InternalTopologyBuilder;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+class StatefulRepartitionNode<K, V, T> extends RepartitionNode<K, V> {
+
+    private final ProcessorSupplier<K, Change<V>> statefulProcessorSupplier;
+    private final MaterializedInternal<K, T, KeyValueStore<Bytes, byte[]>> materialized;
```

Comment:
There's some inconsistency among the stateful nodes here:

1. In `StatefulRepartitionNode` we use `MaterializedInternal`.
2. In `StatefulProcessorNode` we use `StoreBuilder` plus `String[] storeNames`.
3. In `StatefulSourceNode` we use a `StoreBuilder` plus  a `KTableSource`.

Is this intentional? If not, I'd suggest we make it consistent: it seems the `MaterializedInternal` should be sufficient.

Reply:
Yes, this is intentional, as these are the types used for the `InternalTopologyBuilder`.

Reply:
ack, should just be a `ProcessorSupplier`